predicate/src/lib.rs

```
@@ -225,6 +233,140 @@ impl Predicate {
+                    match pp.prune(&summary) {
+                        Ok(matched) => {
+                            assert_eq!(matched.len(), 1);
+                            if matched[0] {
+                                // might match => test next expr
+                                continue;
+                            } else {
+                                // does not match => since expressions are `AND`ed, we know that we will have zero matches
+                                return PredicateMatch::Zero;
+                            }
+                        }
+                        Err(e) => {
+                            debug!(
+                                %e,
+                                %expr,
+                                "cannot prune summary with PruningPredicate",
+                            );
+                            return PredicateMatch::Unknown;
+                        }
+                    }
+                }
+                Err(e) => {
+                    debug!(
+                        %e,
+                        %expr,
+                        "cannot create PruningPredicate from expression",
+                    );
+                    return PredicateMatch::Unknown;
+                }
+            }
+        }
+
+        PredicateMatch::Unknown
+    }
+}
+
+struct SummaryWrapper<'a> {
+    summary: &'a TableSummary,
+}
+
+impl<'a> PruningStatistics for SummaryWrapper<'a> {
+    fn min_values(
+        &self,
+        column: &datafusion::logical_plan::Column,
+    ) -> Option<arrow::array::ArrayRef> {
+        let col = self.summary.column(&column.name)?;
+        let stats = &col.stats;
+
+        // special handling for timestamps
+        if col.influxdb_type == Some(InfluxDbType::Timestamp) {
+            if let data_types::Statistics::I64(val) = stats {
+                return Some(Arc::new(TimestampNanosecondArray::from(vec![val.min])));
+            } else {
+                // wrong data type
```

This is hopefully impossible so could perhaps be a `panic!` but silently returning `None` is also fine (other things are going to explode if a `Timestamp` is not an `i64`).